docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -327,255 +388,102 @@ In Geo-Partitioning, the table must be splitted into partitions, and each partit
-
-DDL on partitions and indices will be discussed below, and other DDL on tables won’t affect placement rules:
-
-* Altering columns
-* Renaming tables
-* Altering charset and collation
-
-### DDL on partitions
-
-TiDB supports adding and dropping partitions.
-
-Once a partition is added, its placement rule is empty and the partition follows the rule of the table it belongs to.
-
-Once a partition is dropped, it can’t be recovered anymore, so its placement rules can be removed immediately.
-
-Also note that DDL on tables may also effect partitions. It's descibed in the section "DDL on tables".
+#### DDL on tables
 
-### DDL on indices
+The placement policy is associated with the definition of the table (and visible in `SHOW CREATE TABLE`). Thus, if a table is recovered by `FLASHBACK` or `RECOVER`, it is expected that the previous rules will be restored.
 
-Once an index is created on an unpartitioned table, the index should follow the rule of the table it belongs to.
+Similarly, `TRUNCATE [TABLE]` does not change the definition of a table. It is expected that as new data is inserted, it will continue to respect placement rules.
 
-Once an index is created on a table with partitions, each part of the index should follow the rule of the partition it belongs to.
+#### SHOW DDL jobs
 
-Once an index is dropped, it can’t be recovered anymore, so its placement rules can be removed immediately.
+All statements related to placement rules must wait until PD returns. If the execution is interrupted, the job will be cancelled and the DDL will rollback, just like other DDL jobs.
 
-Altering primary index is the same with altering secondary indexes. Because if a primary index can be created or dropped, it must be an unclustered index.
-
-Other DDL on indices won’t affect placement rules:
-
-* Renaming index
-* Altering the visibility of index
-
-### Show DDL jobs
-
-As mentioned before, all statements related to placement rules must wait until PD returns. If the execution is interrupted, the job will be cancelled and the DDL will rollback, just like other DDL jobs.
-
-PD schedules regions asynchronously after it returns the message. TiDB can query the progress of scheduling from PD. The progress is observed by executing `SHOW PLACEMENT POLICY` instead of `ADMIN SHOW DDL JOBS`, because the DDL job finishes once PD returns a message.
+PD schedules regions asynchronously after it returns the message. TiDB can query the progress of scheduling from PD. The progress is observed by executing `SHOW PLACEMENT` (or using `information_schema.placement_rules`) instead of `ADMIN SHOW DDL JOBS`, because the DDL job finishes once PD returns a message.
```

Comment:
Assume altering a table's leader placement from "zone+=bj" to "zone+=sh", it can happens that it's leader still remains in bj for a while after ddl execution succeed. How does tidb know that all regions of the table have completed the migration.

Reply:
No progress indicator and easy PD API so far. It will eventually complete, though. Now the only way is to query which store the regions of tables are on, manually.

Reply:
It now says in the proposal a little bit clearer that this is asynchronous, and can be monitored in `SHOW PLACEMENT`.